<?xml version="1.0" ?>
<launch>
  <arg name="STATE_TOPIC_ARM" default="/sim_cartesian_velocity_controller_sim/ee_state"/>
  <arg name="CMD_TOPIC_ARM" default="/sim_cartesian_velocity_controller_sim/command_cart_vel"/>
  <arg name="PID_PARAMS_FILE" default="$(find sim_pid_control)/config/pid_params.yaml" />

  <param name="topic_arm_state" value="$(arg STATE_TOPIC_ARM)"/>
  <param name="topic_arm_command" value="$(arg CMD_TOPIC_ARM) "/>
  <!-- <param name="topic_external_wrench" value="$(arg WRENCH_TOPIC)"/> -->
  
  <arg name="sim" default="false" />

  <rosparam file="$(arg PID_PARAMS_FILE)" command="load"/>

  <!-- <include file="$(find ur5_bringup)/launch/ur5_bringup.launch" >
    <arg name="sim" value="$(arg sim)"/>
    <arg name="F/T_sensor" value="$(arg F/T_sensor)"/>
  </include> -->
    <!-- <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0 0 0 0 0 0 1  world base_link 2" /> -->
  <!-- <rosparam file="$(arg controller_config_file)" command="load"/> -->
    
  <!-- 自己加载的 -->
  <!-- <node name="admittance_controller_manager" pkg="controller_manager" type="controller_manager" respawn="false" 
    output="screen" args="load ur5_cartesian_velocity_controller" /> -->
    <!-- <rosparam file="$(find ur5_bringup)/controllers/joint_state_controller.yaml" command="load"/>
    <node name="joint_state_controller_spawner1" pkg="controller_manager"
      type="controller_manager" args="spawn joint_state_controller" respawn="false" output="screen"/> -->

    <!-- arm controller (to send commands to the arm) Uses the simulated velocity control
      interface to gazebo -->


  <rosparam file="$(find sim_pid_control)/config/sim_vel_controller.yaml" command="load"/>
  <node name="arm_controller_spawner_two" pkg="controller_manager" type="controller_manager" args="load sim_cartesian_velocity_controller_sim" respawn="false" output="screen"/>
  <!-- Make sure to start this in the namespace of the hardware interface -->
  <node pkg="sim_pid_control" name="sim_pid_controller" type="pid_controller_node" output="screen"/>


</launch>
